src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/ReadWriterLocker.cs

```diff
@@ -13,9 +13,13 @@ internal class ReadWriterLocker
     // get another read lock on the same thread
     private readonly ReaderWriterLockSlim _lock = new(LockRecursionPolicy.SupportsRecursion);
 
+    public bool IsReadLockHeld => _lock.IsReadLockHeld;
```

Comment:
We should be consistent on either using `EnsureXXX` or `IsXXX`

Reply:
I don't think those are interchangeable. I expect `Ensure` to actually make state changes (e.g. ensure that some `null` value is initialized), but I wouldn't expect `Is` to do that.

Reply:
I'm onboard with that, but we should rename the current `Ensure` to `Assert` then. Or just remove it and have callers use the `Is`

Reply:
Ah - I see what you're getting at. Yeah, I'll rename that to `Assert`.

Reply:
Actually, there aren't any callers! Will remove.